### PR TITLE
site: exchange Dropzone with Pausly

### PR DESCRIPTION
**Pull request by enyo**
I’m the author of Dropzone.

First of all: thanks a lot for adding me to the showcase! That’s awesome.

I have been working on another project that I think is more fitting though. Pausly is completely written in SvelteKit too (hosted on Vercel, with Supabase as a db) and has been fully migrated to v1.

It uses SSR, has server side authentication support, and uses actions to handle all user interaction. All in all I think it’s a better showcase.

I removed Dropzone so that there are 8 showcases, but if you prefer to keep dropzone in the list, I’m also very happy with that, and can revert the commit where I removed it.

### CODE REVIEW

**Comment by Rich-Harris** [comment]
hey! thanks for the PR — Pausly looks awesome. One of the goals for the showcase is to include sites that have excellent Lighthouse scores — not because we're great believers in Lighthouse (it's a useful diagnostic tool, but it has broken the brains of an entire generation of web developers; a prime example of [Goodhart's law](https://en.wikipedia.org/wiki/Goodhart%27s_law)), but because unfortunately many people judge a framework on the basis of Lighthouse scores without understanding the subtleties behind them.

One of the reasons we were keen to include Dropzone is that it gets excellent Lighthouse scores. Unfortunately Pausly [doesn't score as highly](https://pagespeed.web.dev/report?url=https%3A%2F%2Fwww.pausly.app%2F). One of the major factors appears to be the images like https://pauslyapp.github.io/images/exercises/bad-posture-forward.webp. If these were treated as [imported assets](https://kit.svelte.dev/docs/assets) rather than linked from GitHub Pages, they would be cached immutably. In some cases they could also be made smaller (this is something we'd like to make easier in a future version with better image tooling). 

It's possible that there are other factors that could be addressed (Lighthouse doesn't seem to love `logrocket`, for example) though it's hard to know without working with the source code.

Is there any chance you'd be able to look into some of those Lighthouse recommendations?

**Comment by enyo** [comment]
@Rich-Harris oh yeah sure! I was going to work on that anyway, so if that’s a requirement for the showcase I’m happy to take care of it right away.

**Comment by benmccann** [comment]
@enyo I've written up [some tips for image optimization](https://github.com/sveltejs/kit/issues/241#issuecomment-1274046866), which will automatically build and serve smaller `.avif` images to [browsers that support them](https://caniuse.com/avif) in addition to making them cachable (the caching will help users, but probably not your lighthouse scores). Feel free to let me know if there's any other tips I can provide on setting up image optimization

**Comment by enyo** [comment]
I [got it to 88](https://pagespeed.web.dev/report?url=https%3A%2F%2Fwww.pausly.app%2F). I’ll continue working on the score tomorrow. Of course I’ll try to get as high as possible, but do you have a threshold?

**Comment by benmccann** [comment]
PageSpeed Insights is unfortunately pretty variable , so it's best to run it a few times. I ran it twice and got 69 and 71.

I don't think we have a specific cutoff. I've seen a number of sites get high 90s, but if it were regularly scoring 88 that'd probably be fairly good.

**Comment by enyo** [comment]
@benmccann I’m banding my head against the wall here. I don’t know why I got 88 at one point and now it’s so low. It got it back up to 84, but I’m kind of stuck here.

I tried loading all external JS (like sentry) in an `onMount()` with `await import()` but apparently lighthouse still counts it.

I don’t understand why lighthouse gives the score for all JS loaded, when this is the reason I use SSR in the first place. Do you maybe have a hint here?

If I disable CSR for the landing page, I get a `99` score. Isn’t that the same score I should get when the client is hydrated?

**Comment by Rich-Harris** [comment]
> If I disable CSR for the landing page, I get a `99` score.

This likely indicates that the culprit is either Svelte's hydration, or third party scripts/modules that are imported by your app. The DOM is small (roughly the same size as pages on https://kit.svelte.dev, which Lighthouse views favourably) so my hunch is that third party scripts are to blame. One diagnostic approach could be to just remove `sentry` et al altogether to get a baseline, and then reintroduce them.

It's possible that something like [Partytown](https://partytown.builder.io/) could help, and if so maybe there's stuff we could do to integrate it in a more turnkey fashion.

**Comment by enyo** [comment]
@benmccann I have actually already completely removed sentry. One huge part is now supabase, that I can’t seem to be able to reduce. But I’m getting there.

**Comment by enyo** [comment]
May the performance gods have mercy on me.

image
image

At this point it’s impossible for me to get a lower score, because of supabase. If perf is still an issue here, I’ll need to start working on their clients, because they load too much eagerly and I can’t lazily load it because it determines the user authentication state and I can’t (don’t want to) postpone that.

**Comment by enyo** [comment]
Oh man. I just saw that half the time the score is about 87 or similar. I'll try to continue finding ways to improve that.

**Comment by benmccann** [comment]
Nice! That's a big improvement

Regarding supabase - I wouldn't expect you'd need to load it on the client at all. I'm not exactly sure what page elements you'd fetch from a database. Maybe the "New in Rotation" section? You could create a `+page.server.js` or `+server.js` which fetches from the database and then you'd only need to include the Supabase client on the backend. I'd also expect you could prerender the homepage.

**Comment by enyo** [comment]
@benmccann the reason I need supabase is for authentication. It needs to be loaded initially to maintain the authentication state, and handle things like “magic link” authentication where it will parse the URL and see if there is an authentication token in there.
It also checks that when a user logs in, in another tab, the current tab will be updated to reflect the authentication status as well.
So unfortunately it’s not so easy to remove without making the landing page completely ignorant of authentication state which is something I hope I can avoid.

I can’t completely prerender the page either, because it contains authentication state info. If I’m not mistaken, it wouldn’t help with the performance score either because it will wait for all the JS to be fully downloaded anyway.

What I am currently looking at, is if I can only include supabase if the user is authenticated. Maybe I’ll get somewhere.

**Comment by benmccann** [comment]
Ah, thanks for explaining! I do still think you could probably move it to be used only by an API URL that returns a JSON response to the frontend, which would make it so that it doesn't have to be loaded by the client

**Comment by enyo** [comment]
image

Finally got it consistently over 95

**Comment by enyo** [comment]
I managed to conditionally load supabase on the client when needed. It was a bit tough because so much of the App depends on supabase (for example: there is a store in the layout context, that loads new achievements and subscribes to changes, so they can be shown immediately in an overlay. This now needs to conditionally load only when the user is signed in)

I completely understand why you want the score to be high, but it was mostly useless for the end user because the site is also fully SSR, so the few seconds that can be gained when loading the js don't really impact the user. I wish PageSpeed would take SSR more into account.

Anyway, feel free to merge if you're happy with the score now :)

**Comment by dummdidumm** [comment]
Yeah this is one of the reasons why I think Lighthouse is far from a good tool yet. It says it measures TTI when it actually measures "time to all javascript loaded and initialized".

File: sites/kit.svelte.dev/src/routes/home/Showcase.svelte [hunk 1]
```diff
@@ -4,7 +4,7 @@
 	import pudding from './showcase/pudding.png';
 	import pocketbase from './showcase/pocketbase.png';
 	import pronauns from './showcase/pronauns.png';
-	import dropzone from './showcase/dropzone.png';
```
Comment (benmccann) [inline sites/kit.svelte.dev/src/routes/home/Showcase.svelte:7]: i'll leave it up to rich, but I might personally leave dropzone.png in the repo even if unused. we had talked about creating a kit.svelte.dev/showcase with a larger set of sites and we could include include dropzone there still

**Comment by Rich-Harris** [comment]
amazing work! thank you 🎉

**Pull request merged**